harvester_e2e_tests/integration/test_vm.py

```diff
@@ -0,0 +1,313 @@
+
+    vm_spec = api_client.vms.Spec(1, 1)
+    vm_spec.add_image('disk-0', focal_image['id'])
+    vm_data = []
+    for vm_name in vm_names:
+        code, data = api_client.vms.create(vm_name, vm_spec)
+        assert 201 == code, (
+            f"Failed to create VM {vm_name} with error: {code}, {data}"
+        )
+        vm_data.append(data)
+
+    vmi_data = []
+    vm_name, code, data = None, None, None
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        all_vm_ready = True
+        vmi_data.clear()
+        for vm_name in vm_names:
+            code, data = api_client.vms.get(vm_name)
+            if data.get('status', {}).get('ready', False):
+                code, data = api_client.vms.get_status(vm_name)
+                if code != 200 or data['status']['conditions'][-1]['status'] != 'True':
+                    all_vm_ready = False
+                    break
+                vmi_data.append(data)
+        if all_vm_ready:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Can't find VM {vm_name} with {wait_timeout} timed out\n"
+            f"Got error: {code}, {data}"
+        )
+
+    vm_src_dst_hosts = {}  # {vm_name: [src_host, dst_host]}
+    for vm_name, vm_datum in zip(vm_names, vmi_data):
+        src_host = vm_datum['status']['nodeName']
+        dst_host = next(n for n in available_node_names if n != src_host)
+        vm_src_dst_hosts[vm_name] = [src_host, dst_host]
+
+        code, data = api_client.vms.migrate(vm_name, dst_host)
+        assert code == 204, (
+            f"Can't migrate VM {vm_name} from host {src_host} to {dst_host}",
+            f"Got error: {code}, {data}"
+        )
+
+    code, data = None, None
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        all_vm_migrated = True
+        for vm_name, src_dst_hosts in vm_src_dst_hosts.items():
+            code, data = api_client.vms.get_status(vm_name)
+            if code == 200 and \
+                    data.get('status', {}).get('migrationState', {}).get('completed', False):
+                assert src_dst_hosts[1] == data['status']['nodeName'], (
+                    f"Failed to migrate VM {vm_name} \
+                        from {src_dst_hosts[0]} to {src_dst_hosts[1]}"
+                )
+            else:
+                all_vm_migrated = False
+                break
+        if all_vm_migrated:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"The migration of VM {vm_name} is not completed with {wait_timeout} timed out"
+            f"Got error: {code}, {data}"
+        )
+
+    # teardown
+    for vm_name in vm_names:
+        api_client.vms.delete(vm_name)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
```

Comment:
I would suggest to use `while-for` instead, in `for-while` case, the 2nd VM will never handled if 1st VM timed out.